src/test/java/com/example/demo/controller/BookControllerTest.java

```diff
@@ -0,0 +1,312 @@
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class BookControllerTest {
+    protected static MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+    @Autowired
+    private JdbcTemplate jdbcTemplate;
+
+    @BeforeAll
+    static void beforeAll(
+            @Autowired DataSource dataSource,
+            @Autowired WebApplicationContext applicationContext) throws SQLException {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+        teardown(dataSource);
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/books/insert-books.sql")
+            );
+        }
+    }
+
+    @BeforeAll
+    static void beforeAll(@Autowired CategoryRepository categoryRepository) {
+        categoryRepository.save(new Category("Fantasy"));
+    }
+
+    @AfterAll
+    static void afterAll(
+            @Autowired DataSource dataSource
+    ) {
+        teardown(dataSource);
+    }
+
+    @SneakyThrows
+    private static void teardown(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/books/cleanup-db.sql")
+            );
+        }
+    }
+
+    @WithMockUser(username = "admin@mail.com", roles = {"ADMIN"})
+    @Test
+    @Sql(scripts = "classpath:database/books/cleanup-db.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    void createBook_ValidRequestDto_Success() throws Exception {
+        // given
+        CreateBookRequestDto createBookRequestDto = new CreateBookRequestDto()
+                .setAuthor("Joseph Heller")
+                .setPrice(BigDecimal.valueOf(15.99))
+                .setTitle("Catch-22")
+                .setIsbn("12324564789")
+                .setDescription("Great novel.")
+                .setCategories(List.of());
+
+        BookDto expected = new BookDto()
+                .setId(1L)
+                .setAuthor(createBookRequestDto.getAuthor())
+                .setPrice(createBookRequestDto.getPrice())
+                .setTitle(createBookRequestDto.getTitle())
+                .setIsbn(createBookRequestDto.getIsbn())
+                .setDescription(createBookRequestDto.getDescription())
+                .setCategories(createBookRequestDto.getCategories());
```

Comment:
better move object creation to the private method or util class and simplify your test cases